src/sb-util.ts

```diff
@@ -61,13 +61,12 @@ export class ScratchProject {
 
     public blocks(): BlockCollection {
         // Getting an iterable collection of block properties from all the sprites
-        const blocks =
-            // A transformation from an Iterable<Sprite> to Iterable<BlockProperties>
-            makeIterable(this.sprites(), function*(sprites): Iterator<BlockProperties> {
-                for (const sprite of sprites) {
-                    yield* sprite.blocks().propsIterable();
-                }
-            });
+        const iter = function*(sprites): Iterator<BlockProperties> {
+            for (const sprite of sprites) {
+                yield* sprite.blocks().propsIterable();
+            }
+        };
+        const blocks = new MemoizedIterable(iter(this.sprites()));
```

Comment:
I'm not 100% sure we should swap over to being memoized by default -- this is a further discussion we should have and leave as documentation in the readme I think though... Using a memoized search gives us this "just in time but kind of frozen" view of our queries. However:

* Do we want our queries to be "live search" such that if you edit the blocks it would find the new one in an old collection? (document.querySelectorAll() has similar behaviors)

* Or, do we want to have to opt in, I.E. `this.sprites().memo().blocks()` so the sprite search is locked in but the blocks arent

* Or do we want to have it always be "search again if you want newest results" which was more the approach that jQuery took with it's collections

-----

Mind starting this discussion / weighing in on it with opinions?